tiledb/sm/c_api/tiledb_filestore.cc

```diff
@@ -272,158 +294,194 @@ int32_t tiledb_filestore_uri_import(
        static_cast<uint64_t>(file_size % tile_extent != 0)) *
           tile_extent -
       1;
-  subarray.add_range(0, static_cast<uint64_t>(0), last_space_tile_boundary);
+  uint64_t range_arr[] = {static_cast<uint64_t>(0), last_space_tile_boundary};
+  tiledb::type::Range subarray_range(
+      static_cast<void*>(range_arr), sizeof(uint64_t) * 2);
+  throw_if_not_ok(subarray.add_range(0, std::move(subarray_range)));
   query.set_subarray(subarray);
 
-  auto tiledb_cloud_fix = [&](uint64_t start, uint64_t end, uint64_t nbytes) {
-    Query query(context, array);
-    query.set_layout(TILEDB_ROW_MAJOR);
-    Subarray subarray(context, array);
-    subarray.add_range(0, start, end);
-    query.set_subarray(subarray);
-    query.set_data_buffer(
-        tiledb::sm::constants::filestore_attribute_name, buffer.data(), nbytes);
-    query.submit();
+  auto tiledb_cloud_fix = [&](uint64_t start, uint64_t end) {
```

Comment:
Please create a followup task for switching filestore API to using global order writes natively for tiledb URIs.

Reply:
https://app.shortcut.com/tiledb-inc/story/36254/filestore-api-should-use-global-order-writes